tests/compatibility-test.py

```diff
@@ -54,10 +58,11 @@ def create_cluster():
 
 
 def apply_kuberay_resources():
-    shell_assert_success('kind load docker-image kuberay/operator:{}'.format(kuberay_sha))
-    shell_assert_success('kind load docker-image kuberay/apiserver:{}'.format(kuberay_sha))
-    shell_assert_success(
-        'kubectl create -k manifests/cluster-scope-resources')
+    shell_assert_success('kind load docker-image {}'.format(kuberay_operator_image))
```

Comment:
we need to push the correct build to kind cluster. not the nightly build.

Reply:
The variable `kuberay_operator_image` will be updated in the function `parse_environment()`. Hence, the value of `kuberay_operator_image` is correct.

The previous CI errors are caused by `docker pull`. The new image builds (e.g. `kuberay/operator:1ebf6a8`) only exist in the local docker registry, and thus errors will be reported when we try to pull `kuberay/operator:1ebf6a8` from DockerHub. Hence, I updated the function `download_images` to pull images only when the images do not exist in the local docker registry.